nixos/modules/services/misc/forgejo.nix

```diff
@@ -322,6 +322,12 @@ in
                 description = "Disable external SSH feature.";
               };
 
+              START_SSH_SERVER = mkOption {
+                type = types.bool;
+                default = false;
+                description = "When enabled, use the built-in SSH server.";
+              };
```

Comment:
I hesitate whether we need that option right now.
`services.forgejo.settings` is already a free-form option, so we technically don't need it.

The only benefit would be self-documentation. Making users aware that one does not have to use the hosts' openssh server.

What is your rationale behind adding it? :)

For the integrated SSH server to be able to bind to port < 1024, we need to add a few lines to `systemd.services.forgejo.serviceConfig`.

Namely:
```diff
diff --git a/nixos/modules/services/misc/forgejo.nix b/nixos/modules/services/misc/forgejo.nix
index babed2d5acd4..c8aada68e4a5 100644
--- a/nixos/modules/services/misc/forgejo.nix
+++ b/nixos/modules/services/misc/forgejo.nix
@@ -616,6 +616,10 @@ in
         # System Call Filtering
         SystemCallArchitectures = "native";
         SystemCallFilter = [ "~@cpu-emulation @debug @keyring @mount @obsolete @privileged @setuid" "setrlimit" ];
+      } // lib.optionalAttrs (cfg.settings.server.SSH_PORT < 1024) {
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
+        PrivateUsers = false;
       };
 
       environment = {
```

Reply:
My rationale behind adding the option to the documentation is that it's utilized inside the module itself. Since it's part of that, I feel that it's a good idea to also have it in the documentation, in case someone doesn't look at the upstream's configuration documentation. While that's not something you should do, it's bound to happen, and so I feel that the module should have documentation for at least every option that it uses internally.